packages/connect/src/device/checkFirmwareRevision.ts

```diff
@@ -0,0 +1,71 @@
+import { PROTO } from '../constants';
+import { downloadReleasesMetadata } from '../data/downloadReleasesMetadata';
+import { FirmwareRelease, VersionArray } from '../types';
+import { DeviceModelInternal, FirmwareRevisionCheckResult } from '../types/device';
+
+type GetOnlineReleaseMetadataParams = {
+    deviceVersion: VersionArray;
+    internalModel: string;
+};
+
+const getOnlineReleaseMetadata = async ({
+    deviceVersion,
+    internalModel,
+}: GetOnlineReleaseMetadataParams): Promise<FirmwareRelease | undefined> => {
+    const onlineReleases = await downloadReleasesMetadata({ internal_model: internalModel });
+
+    return onlineReleases.find(onlineRelease => onlineRelease.version === deviceVersion);
+};
+
+const failFirmwareRevisionCheck = (
+    error: FirmwareRevisionCheckResult['error'],
+): FirmwareRevisionCheckResult => ({ success: false, error });
+
+export type CheckFirmwareRevisionParams = {
+    deviceVersion: VersionArray | [];
+    internalModel: PROTO.DeviceModelInternal;
+    deviceRevision: string | null;
+    expectedRevision: string | undefined;
+};
+
+export const checkFirmwareRevision = async ({
+    deviceVersion,
+    internalModel,
+    deviceRevision,
+    expectedRevision,
+}: CheckFirmwareRevisionParams): Promise<FirmwareRevisionCheckResult> => {
+    if (internalModel !== DeviceModelInternal.T2T1 && internalModel !== DeviceModelInternal.T1B1) {
+        return { success: true, error: null }; // For newer devices we are covered by secure element anyway
+    }
+
+    deviceRevision = 'yolo'; // TODO: remove this!!!
+
+    if (expectedRevision === undefined) {
+        if (deviceVersion.length !== 3) {
+            return failFirmwareRevisionCheck('firmware-version-unknown');
+        }
+
+        try {
+            const onlineRelease = await getOnlineReleaseMetadata({ deviceVersion, internalModel });
```

Comment:
sooo, there is no local first variant?

Reply:
Yes there is. It is provided into the function through params from here:

![image](https://github.com/user-attachments/assets/b1b543eb-24b3-4940-ac6b-7b755d726feb)

Reply:
optimiziation idea: 
maybe we could pull loading of the latest list of releases somewhere higher so that it doesn't block device features update 🤔

Reply:
plus we can download them once and then update current releases stored in DataManager (from what i understand it would fetch new releases on each device connect event)

Reply:
I like the idea, but this shall happen only rarely (newer firmware in older suite). I expect this to be a super-edge-case and its not worth it to make more complicated code just to address it. WDYT?